06_neighborhoods/Transportation/transit_trips_cost_county_subgroups.R

```diff
@@ -0,0 +1,645 @@
+# Calculate percentile ranks
+trips_all_15$percentile_rank <- (trips_all_15$rank - 1) / (nrow(trips_all_15) - 1) * 100
+trips_all_19$percentile_rank <- (trips_all_19$rank - 1) / (nrow(trips_all_19) - 1) * 100
+# round to 2 decimals
+trips_all_15$index_transit_trips <- round(trips_all_15$percentile_rank, 2)
+trips_all_19$index_transit_trips <- round(trips_all_19$percentile_rank, 2)
+# keep only what we need
+trips_all_15 <- trips_all_15 %>%
+  select(state, county, index_transit_trips, subgroup_type, subgroup)
+trips_all_19 <- trips_all_19 %>%
+  select(state, county, index_transit_trips, subgroup_type, subgroup)
+
+
+# 3b. Collapse accordingly -- to counties and race categories, weighting the measure by HH count per tract
+
+cost_by_race_15 <- cost_data_2015 %>%
+  group_by(state, county, race_category) %>%
+  summarize(transit_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE))
+
+cost_by_race_19 <- cost_data_2019 %>%
+  group_by(state, county, race_category) %>%
+  summarize(transit_cost = weighted.mean(x = t_80ami, w = households, na.rm = TRUE))
+
+trips_by_race_15 <- trips_data_2015 %>%
+  group_by(state, county, race_category) %>%
+  summarize(transit_trips = weighted.mean(x = transit_trips_80ami, w = households, na.rm = TRUE))
+
+trips_by_race_19 <- trips_data_2019 %>%
+  group_by(state, county, race_category) %>%
+  summarize(transit_trips = weighted.mean(x = transit_trips_80ami, w = households, na.rm = TRUE))
+
+# Make sure we have 3 race vars accounted for each county - create dummy df for merging purposes
+county_expander <- expand_grid(
+  count(cost_by_race_15, state, county) %>% select(-n),
+  race_category = c("Majority White-NH Tracts", "Majority Non-White Tracts", "Mixed Race and Ethnicity Tracts")
+)
+
+# merge all four datasets into this dummy to account for all race categories
+cost_by_race_15 <- left_join(county_expander, cost_by_race_15, by = c("state", "county", "race_category")) %>%
+  mutate(subgroup_type = "race-ethnicity") %>%
+  rename(subgroup = race_category,
+         index_transportation_cost = transit_cost)
+# 9429 obs = 3*3143 counties -> good to go
+
+cost_by_race_19 <- left_join(county_expander, cost_by_race_19, by = c("state", "county", "race_category")) %>%
+  mutate(subgroup_type = "race-ethnicity") %>%
+  rename(subgroup = race_category,
+         index_transportation_cost = transit_cost)
+
+trips_by_race_15 <- left_join(county_expander, trips_by_race_15, by = c("state", "county", "race_category")) %>%
+  mutate(subgroup_type = "race-ethnicity") %>%
+  rename(subgroup = race_category)
+
+trips_by_race_19 <- left_join(county_expander, trips_by_race_19, by = c("state", "county", "race_category")) %>%
+  mutate(subgroup_type = "race-ethnicity") %>%
+  rename(subgroup = race_category)
+
+# remove dummy df
+rm(county_expander)
+
+
+
+# CHANGE THE TRIPS VALUES TO NATIONAL PERCENTILE RANKS FOR EACH
+# For 2015 data
+trips_by_race_15 <- trips_by_race_15 %>%
```

Comment:
Maybe I am missing something but a lot of NAs are getting ranked here too?

Reply:
Yes, thank you for catching! Updated to exclude NA values in the ranking